eloquentjs/5_3_everything.js

```diff
@@ -0,0 +1,83 @@
+function everythingSome(inputArray, testFunction) {
+    const antiTest = j => !testFunction(j);
+    return inputArray.some(testFunction) && !inputArray.some(antiTest);
+}
+
+function everythingLoop(inputArray, testFunction) {
+    for (let element of inputArray) {
+        if (!testFunction(element)) {
+            return false
+        }
+    }
+    return true;
+}
+
+const greaterThanN = n => (x => x > n);
+const greaterThanZero = greaterThanN(0);
+const greaterThanTwo = greaterThanN(2);
+const greaterThanTen = greaterThanN(10);
+const greaterThanNegativeFifty = greaterThanN(-50);
+const greaterThanFifty = greaterThanN(50);
+
+const testCases = [
+    [0, 1, 2, 3],
+    [1, 2, 3],
+    [-2, -1, 0, 1, 2],
+    [-2, -1, 0],
+    [-2, -1],
+    [0],
+    [10, 11, 12],
+    [-12, -11, -10],
+    [65, -766, 432, -1, 8],
+];
+
+const testFunctionDictionary = {
+    greaterThanZero: greaterThanZero,
+    greaterThanTwo: greaterThanTwo,
+    greaterThanTen: greaterThanTen,
+    greaterThanNegativeFifty: greaterThanNegativeFifty,
+    greaterThanFifty: greaterThanFifty,
+};
+
+console.log("\nTesting Everything Some...\n");
+testEverythingFunction(everythingSome, testCases, testFunctionDictionary);
+
+console.log("Testing Everything Loop...\n");
+testEverythingFunction(everythingLoop, testCases, testFunctionDictionary);
+
+function testEverythingFunction(everythingFunc, testCases, testFunctionDictionary) {
+    let failureStatements = [];
```

Comment:
This is better for sure, but the thing I'm trying to get you ready for is the need to be able to write a little testing setup like this in 3 minutes tops in an interview. Let me show you an example of a more minimal setup:

```js
function testEverythingFunction(everythingFunc) {
    for (const testArray of testCases) {
        for (const [name, func] of Object.entries(testFunctionDictionary)) {
            if (everythingResult !== testArray.every(func))) {
                console.log(`\nFailure: ${name} ${testArray}`);
            else {
                 console.log('.');
            }
        }
    }
}
```

This is about as minimal as you can make it while still having it do everything it needs to do, and it's a general pattern that works very well in certain sorts of interviews.

Also, it's worth noting that an issue with your solution that this one shares and is fine in this case but you should know about is that if a test case causes the program to crash it will swallow the error. You can solve that issue with a try catch block, but there's no need to do it here or in an interview.

Reply:
This makes sense to me. Accomplishes the same thing w/o unnecessary complexity. What's the justification for             ```
else {
                 console.log('.');
            }
```
Or is it just a placeholder? Also, was the dictionary the best approach here to grabbing both the function names and the functions themselves? I initially tried using an array but couldn't grab the names using `func.name` because of them being anonymous.

Reply:
The reason for that else clause is to print something on success. That way you know the tests are running and that they succeeded.

I'd recommend moving away from prose in your test outputs. The words muddle the important content and add time to the implementation.

Using the map was totally fine, yeah. You could use an array of arrays or objects and de structure them if you wanted - I think that's what I'd do - but it's not necessarily better.